Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -796,11 +813,107 @@ namespace xr
+                ArPlane* subsumingPlane = nullptr;
+                ArPlane_acquireSubsumedBy(session, planeTrackable, &subsumingPlane);
+                if (subsumingPlane != nullptr)
+                {
+                    ArTrackable_release(trackable);
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(subsumingPlane));
+                    continue;
+                }
+
+                // Get the center pose.
+                float rawPose[7]{};
+                ArPlane_getCenterPose(session, planeTrackable, tempPose);
+                ArPose_getPoseRaw(session, tempPose, rawPose);
+
+                // Dynamically allocate the polygon array, and fill it in.
+                int32_t polygonSize;
+                ArPlane_getPolygonSize(session, planeTrackable, &polygonSize);
+                float* polygon = reinterpret_cast<float*>(malloc(sizeof(float) * polygonSize));
+
+                // Get the polygon.
+                ArPlane_getPolygon(session, planeTrackable, polygon);
+
+                // Update the existing plane if it exists, otherwise create a new plane, and add it to our list of planes.
+                auto planeIterator = existingPlanes.find(reinterpret_cast<NativePlaneIdentifier>(trackable));
+                if (planeIterator != existingPlanes.end())
+                {
+                    UpdateExistingPlane(planeIterator->second, rawPose, polygon, polygonSize);
+                }
+                else
+                {
+                    // This is a new plane, create it and initialize its values.
+                    Plane plane{};
+                    plane.NativePlaneId = reinterpret_cast<NativePlaneIdentifier>(trackable);
+                    UpdateExistingPlane(&plane, rawPose, polygon, polygonSize);
+                    newPlanes.push_back(plane);
+                }
+            }
+        }
+
+        bool CheckIfPlaneWasUpdated(Plane* existingPlane, float* newPolygon, size_t newPolygonSize, Pose& newCenter)
+        {
+            // First check if the center has changed, or the polygon size has changed.
+            if (existingPlane->Center != newCenter
+                || existingPlane->PolygonSize * 2 != newPolygonSize)
+            {
+                return true;
+            }
+
+            // Next loop over the polygon and check if any points have changed.
+            for (size_t i = 0; i < newPolygonSize; i++)
+            {
+                if (abs(existingPlane->Polygon[i] - newPolygon[i]) > FLOAT_COMPARISON_THRESHOLD)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        // Cleans up a plane. Called when a plane gets deleted.
+        void CleanupPlane(Plane* plane)
+        {
+            ArTrackable_release(reinterpret_cast<ArTrackable*>(plane->NativePlaneId));
+
+            if (plane->Polygon != nullptr)
+            {
+                free(plane->Polygon);
```

Comment:
Similar comment as above, why `free` (paired with `malloc`) instead of `delete` (paired with `new`)?

Reply:
Got rid of this manual memory management.